### PR TITLE
fixes #3151 by skipping the lint instead of crashing

### LINKED ISSUE #3151: 'rustc' panicked at 'failed to create span for type arguments'
Hi,
this code leads to a panic in a clippy lint:

```rust
#[derive(Clone)]
pub struct HashMap<V, S> {
    hash_builder: S,
    table: RawTable<V>,
}

#[derive(Clone)]
pub struct RawTable<V> {
    size: usize,
    val: V
}
```

I did put the code in a small repo:

    git clone https://github.com/PSeitz/span_err

Stacktrace:

```
thread 'rustc' panicked at 'failed to create span for type arguments', libcore/option.rs:1000:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: rustc::util::common::panic_hook
   5: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:481
   6: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:391
   7: rust_begin_unwind
             at libstd/panicking.rs:326
   8: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   9: core::option::expect_failed
             at libcore/option.rs:1000
  10: <clippy_lints::types::ImplicitHasher as rustc::lint::LateLintPass<'a, 'tcx>>::check_item
  11: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_item
  12: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_mod
  13: rustc::hir::intravisit::walk_crate
  14: rustc::lint::context::check_crate
  15: rustc::util::common::time
  16: rustc::ty::context::tls::enter_context
  17: <std::thread::local::LocalKey<T>>::with
  18: rustc::ty::context::TyCtxt::create_and_enter
  19: rustc_driver::driver::compile_input
  20: rustc_driver::run_compiler_with_pool
  21: <scoped_tls::ScopedKey<T>>::set
  22: rustc_driver::run_compiler
  23: <scoped_tls::ScopedKey<T>>::set
```

This block seems to be the issue:
https://github.com/rust-lang-nursery/rust-clippy/blob/fdd830f52c082b83db0dac3e0066c0cf114050d2/clippy_lints/src/types.rs#L1811-L1816

Related, but seems to be a different error cause (no check_item in the stacktrace): rust-lang-nursery/rust-clippy#2306

### PR DESCRIPTION
**Pull request by PSeitz**

### CODE REVIEW

**Comment by PSeitz** [comment]
the dogfood error is unrelated, like in other recent pull requests #3149 #3148

File: clippy_lints/src/types.rs [hunk 1]
```diff
@@ -1810,9 +1810,12 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ImplicitHasher {
 
                     let generics_suggestion_span = generics.span.substitute_dummy({
                         let pos = snippet_opt(cx, item.span.until(target.span()))
-                            .and_then(|snip| Some(item.span.lo() + BytePos(snip.find("impl")? as u32 + 4)))
-                            .expect("failed to create span for type arguments");
-                        Span::new(pos, pos, item.span.data().ctxt)
+                            .and_then(|snip| Some(item.span.lo() + BytePos(snip.find("impl")? as u32 + 4)));
+                        if let Some(pos) = pos {
+                            Span::new(pos, pos, item.span.data().ctxt)
+                        }else{
```
Comment (oli-obk) [inline clippy_lints/src/types.rs]: Can you add some spaces here?

Comment (PSeitz) [reply clippy_lints/src/types.rs]: ok, done. `cargo fmt` didn't change this, maybe we should adjust rustfmt.toml?

Comment (oli-obk) [reply clippy_lints/src/types.rs]: weird, not sure why it wouldn't pick up on that.

**Comment by Manishearth** [comment]
Hi! Could you leave the requested comment in https://github.com/rust-lang-nursery/rust-clippy/issues/3230 for relicensing?

**Pull request merged**